go/mode_docs.bzl

```diff
@@ -0,0 +1,200 @@
+May also be set with the `--strip` command line option, which affects C/C++ targets, too.
+
+#### `debug`
+- Optional boolean
+- Default: `False`
+
+Includes debugging information in compiled packages (using the `-N` and `-l` flags).
+
+#### `gotags`
+- Optional list of strings
+- Default: `[]`
+
+Controls which build tags are enabled when evaluating build constraints in source files. Useful for conditional compilation.
+
+
+### `linkmode`
+- Optional string
+- Default: `"normal"`
+
+Determines how the Go binary is built and linked. Similar to `-buildmode`. 
+Must be one of `"normal"`, `"shared"`, `"pie"`, `"plugin"`, `"c-shared"`, `"c-archive"`.
+
+
+Platforms
+---------
+
+You can define a [Bazel platform] using the native [platform] rule. A platform
+is essentially a list of facts (constraint values) about a target platform.
+rules_go defines a `platform` for each configuration the Go toolchain supports
+in `@io_bazel_rules_go//go/toolchain`. There are also [config_setting] targets
+in `@io_bazel_rules_go//go/platform` that can be used to pick platform-specific
+sources or dependencies using [select].
+
+You can specify a target platform using the `--platforms` command line flag.
+Bazel will automatically select a registered toolchain compatible with the
+target platform (rules_go registers toolchains for all supported platforms).
+For example, you could build for Linux / arm64 with the flag
+`--platforms=@io_bazel_rules_go//go/toolchain:linux_arm64`.
+
+You can set the `goos` and `goarch` attributes on an individual
+[go_binary] or [go_test] rule to build a binary for a specific platform.
+This sets the `--platforms` flag via [Bazel configuration transitions].
+
+
+Examples
+--------
+
+### Building pure go binaries
+
+You can switch the default binaries to non cgo using
+
+``` bash
+bazel build --@io_bazel_rules_go//go/config:pure //:my_binary
+```
+
+You can build pure go binaries by setting those attributes on a binary.
+
+``` bzl
+go_binary(
+    name = "foo",
+    srcs = ["foo.go"],
+    pure = "on",
+)
+```
+
+
+### Building static binaries
+
+| Note that static linking does not work on darwin.
+
+You can switch the default binaries to statically linked binaries using
+
+``` bash
+bazel build --@io_bazel_rules_go//go/config:static //:my_binary
+```
+
+You can build static go binaries by setting those attributes on a binary.
+If you want it to be fully static (no libc), you should also specify pure.
+
+``` bzl
+go_binary(
+    name = "foo",
+    srcs = ["foo.go"],
+    static = "on",
+)
+```
+
+### Using the race detector
+
+You can switch the default binaries to race detection mode, and thus also switch
+the mode of tests by using
+
+``` bash
+bazel test --@io_bazel_rules_go//go/config:race //...
+```
+
+Alternatively, you can activate race detection for specific tests.
+
+``` bzl
+go_test(
+    name = "go_default_test",
+    srcs = ["lib_test.go"],
+    embed = [":go_default_library"],
+    race = "on",
+)
+```
+"""
```

Comment:
Can you help me understand the value of this `.bzl` file? It doesn't seem that Stardoc is actually adding value here. This file is just directly being projected into the `.md` file without any generated components. Is there a way to get any of these fields out of the `.bzl` files instead of manually documenting them?

Reply:
Closing for now, since we don't have a way to support generating documentation within BUILD files. We would have to add support for this within stardoc.